testdata/p4_16_samples/forloop1.p4

```diff
@@ -0,0 +1,32 @@
+#include <core.p4>
+control generic<M>(inout M m);
+package top<M>(generic<M> c);
+
+header t1 {
+    bit<32>     f1;
+    bit<16>     h1;
+    bit<8>      b1;
+    bit<8>      cnt;
+}
+
+header t2 {
+    bit<32>     x;
+}
+
+struct headers_t {
+    t1          head;
+    t2[8]       stack;
+}
+
+control c(inout headers_t hdrs) {
+    apply {
+        for (t2 v in hdrs.stack) {
```

Comment:
@asl Any reason for `in` as opposed to C++ style foreach `:`? This way we can avoid introducing a new keyword. Is this emulating Rust-style? Looks a little odd with the type specifier.

Reply:
Downstream we used `foreach (type foo in range)` syntax, but @ChrisDodd changed to `for` :) 

My idea was to use entirely new keyword to explicitly deviate from C / C++ syntax to outline semantic differences. I'm fine with C++-style range for syntax just in case.

Reply:
`in` is already keyword in P4, so does not constitute a new keyword, just repurposing an existing one.  Using `for` instead of `foreach` is avoiding a new keyword, and is what was in Andy's original change.